### PR TITLE
Added wld_removenpc call

### PR DESCRIPTION
**Pull request by Domarion**
Wld_RemoveNpc(instance) removes NPC from the world. This function is used by [Gothic2 Notr unofficial patch](https://github.com/dosinabox/g2nr_unofficial_update/blob/20e506c5ed07a2b915e96f93661d39466afe568b/PrjGOTHIC/Story/Events/B_GolemChest.d#L10).

Also Wld_RemoveNpc is present in some scripting tutorials and in Gothic.DAT file for Gothic1 1.08, Gothic2 Notr.

### CODE REVIEW

**Review by Try — COMMENTED** [review]
Hi, @Domarion and thanks for PR!

I've left some review comments bellow. Also red-build - it's on me, master should be green now

File: game/world/worldobjects.cpp [hunk 1]
```diff
@@ -460,6 +460,17 @@ Npc *WorldObjects::findNpcByInstance(size_t instance, size_t n) {
   return nullptr;
   }
 
+void WorldObjects::removeNpcByInstance(size_t instance) {
+  for(size_t i=0; i<npcArr.size(); ++i){
+    auto& npc=*npcArr[i];
+    if(npc.handle().symbol_index()==instance){
```
[the remark this answers is not in the document]
Comment (Domarion) [reply game/world/worldobjects.cpp:466]: I got little information about `wld_removenpc(instance)` in vanilla engine. This function cannot be used when there are global references "self, other, victim", instance must be unique, it doesn't add npc to temporary limbo, completely removes npc from memory without checks.

[the remark this answers is not in the document]
Comment (Try) [reply game/world/worldobjects.cpp:466]: Removing npc without check is definitely not an option, as it enables for script to have use-after-free error and crash.

Assuming that script is totally correct and never uses npc, after deleting it, than limbo/delete wont be possible to detect.
If script uses instance (by accident), than engine will allow to essentially observe last known state of the npc.

**Comment by Try** [comment]
Implemented `wld_removenpc` separately, as there is no progress in this PR

**Pull request closed without merging**